### PR TITLE
Teams screen

### PR DESCRIPTION
**Pull request by pmozola**

### CODE REVIEW

File: src/API/FL.API/Controllers/TeamController.cs [hunk 1]
```diff
@@ -35,5 +36,23 @@ public async Task<ActionResult> Get()
                await this.mediator
                     .Send(new GetTeamListQuery()));
         }
+
+        [HttpGet("top")]
+        [AllowAnonymous]
+        public async Task<ActionResult> TopFive()
```
Comment (treformat) [inline src/API/FL.API/Controllers/TeamController.cs]: I would add new parameter (count/amount) to GetTopQuery without hardcoded 5

File: src/API/FL.API/Controllers/TeamController.cs [hunk 2]
```diff
@@ -35,5 +36,23 @@ public async Task<ActionResult> Get()
...
+                    .Send(new GetTopFiveTeamQuery()));
+        }
+
+        [HttpGet("{name}")]
+        [AllowAnonymous]
+        public async Task<ActionResult> FilterByName(string name)
+        {
+            return this.Ok(
+                await this.mediator
+                    .Send(new GetTeamByNameQuery(name)));
```
Comment (treformat) [inline src/API/FL.API/Controllers/TeamController.cs]: Find instead of Get

File: src/API/FL.API/Queries/Database/LeagueReadModelContext.cs [hunk 3]
```diff
@@ -16,5 +16,7 @@ public LeagueReadModelContext(DbContextOptions<LeagueReadModelContext> options)
         public DbSet<DivisionViewModel> Division { get; set; }
 
         public DbSet<TeamViewModel> Team { get; set; }
+
+        public DbSet<DivisionTeamViewModel> DivisionTeam { get; set; }
```
Comment (treformat) [inline src/API/FL.API/Queries/Database/LeagueReadModelContext.cs]: Default EF conventions use plural form of properties like, Teams, DevisionTemas

File: src/API/FL.API/Queries/QueryHandlers/GetTopFiveTeamQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        private readonly LeagueReadModelContext repository;
+
+        public GetTopFiveTeamQueryHandler(LeagueReadModelContext repository)
+        {
+            this.repository = repository;
+        }
+
+        public async Task<List<TeamViewModel>> Handle(GetTopFiveTeamQuery request, CancellationToken cancellationToken)
+        {
+            return await this.repository.Team.Take(5).ToListAsync(cancellationToken);
```
Comment (treformat) [inline src/API/FL.API/Queries/QueryHandlers/GetTopFiveTeamQueryHandler.cs]: Hardcoded 5

File: src/API/FL.API/Queries/ViewModels/TeamViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+
+namespace FL.API.Queries.ViewModels
+{
+    public class TeamViewModel
+    {
+        public TeamViewModel()
```
Comment (treformat) [inline src/API/FL.API/Queries/ViewModels/TeamViewModel.cs]: Do you need parameterless constructor?

File: src/API/FL.Application/FL.Application/CommandHandlers/Division/AddNewTeamToDivisionCommandHandler.cs [hunk 6]
```diff
@@ -11,12 +11,10 @@
     public class AddNewTeamToDivisionCommandHandler : AsyncRequestHandler<AddNewTeamToDivisionCommand>
     {
         private readonly IRepository<Season> repository;
-        private readonly IMediator mediator;
 
-        public AddNewTeamToDivisionCommandHandler(IRepository<Season> repository, IMediator mediator)
+        public AddNewTeamToDivisionCommandHandler(IRepository<Season> repository)
```
Comment (treformat) [inline src/API/FL.Application/FL.Application/CommandHandlers/Division/AddNewTeamToDivisionCommandHandler.cs]: Assign

File: src/API/FL.Domain/Aggregates/TeamAggregate/Team.cs [hunk 7]
```diff
@@ -1,27 +1,45 @@
...
     public class Team : AggregateRoot
     {
-        public Team(string name, string email)
+        public Team(string firstName, string lastName, string email)
         {
-            this.Name = name;
-            this.IsActive = true;
-            this.EmailAddress = new EmailAddress(email);
-            base.Id = new Identity();
+            if (string.IsNullOrEmpty(firstName))
```
Comment (treformat) [inline src/API/FL.Domain/Aggregates/TeamAggregate/Team.cs]: I prefer using single Set method for one property like: SetName or ChangeName and put validation and assign there. It makes constructor cleaner and if in future you would like to change name in some other way (like update) you make only this method public.

File: src/API/FL.Domain/Aggregates/TeamAggregate/Team.cs [hunk 8]
```diff
@@ -1,27 +1,45 @@
...
+﻿namespace FL.Domain.Aggregates.TeamAggregate
 {
+    using System;
+    using FL.Domain.Aggregates.TeamAggregate.Events;
+    using FL.Domain.BaseObjects;
+
     public class Team : AggregateRoot
     {
-        public Team(string name, string email)
+        public Team(string firstName, string lastName, string email)
```
Comment (treformat) [inline src/API/FL.Domain/Aggregates/TeamAggregate/Team.cs]: TeamAggreage should implements interfaces for event handling? IEventHandler<TeamCreatedDomainEvent> ?

File: src/API/FL.Infrastructure.Database/AggregateRepository.cs [hunk 9]
```diff
@@ -14,10 +15,12 @@ public class AggregateRepository<T> : IRepository<T>
         where T : AggregateRoot, new()
     {
         private readonly AggregateStoreContext context;
+        private readonly IMediator mediator;
 
-        public AggregateRepository(AggregateStoreContext context)
+        public AggregateRepository(AggregateStoreContext context, IMediator mediator)
```
Comment (treformat) [inline src/API/FL.Infrastructure.Database/AggregateRepository.cs]: I thing that aggregate should't has reference for database context and mediator. I would put it into repository. So in aggregate you only add events to some event list (unpublished changes) and during executing Save method of some repository implementation you should get this list and publish events.

File: src/Client/src/app/settings/pages/divisions/add-team-to-division-dialog/add-team-to-division-dialog.component.html [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+        <mat-option *ngFor="let option of filteredOptions | async" [value]="option.id">
+          {{option.firstName}}
+        </mat-option>
+      </mat-autocomplete>
+    </mat-form-field>
+  </form>
+</div>
+<div mat-dialog-actions>
+  <button mat-button (click)="onSubmit()" [disabled]="!teamForm.valid" tabindex="2">Zapisz</button>
+  <button mat-button (click)="onNoClick()" tabindex="-1">Anuluj</button>
```
Comment (treformat) [inline src/Client/src/app/settings/pages/divisions/add-team-to-division-dialog/add-team-to-division-dialog.component.html]: onNoClick? Just onCancel

File: src/Client/src/app/settings/pages/divisions/add-team-to-division/add-team-to-division.component.css [hunk 11]
```diff
@@ -0,0 +1,3 @@
+table {
```
Comment (treformat) [inline src/Client/src/app/settings/pages/divisions/add-team-to-division/add-team-to-division.component.css]: i would add this to global css for example as ".full-width" class and use it in html on table element.

File: src/Client/src/app/settings/pages/divisions/divisions.component.ts [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+  templateUrl: './divisions.component.html',
+  styleUrls: ['./divisions.component.css']
+})
+export class DivisionsComponent implements OnInit {
+  divisions: Observable<IDivision[]>;
+
+  constructor(private divisionService: DivisionService) { }
+
+  ngOnInit() {
+    this.divisions = this.divisionService.getDivisions();
```
Comment (treformat) [inline src/Client/src/app/settings/pages/divisions/divisions.component.ts]: getDvisions return observable, missing subscribe?

File: src/Client/src/app/settings/pages/teams/dialog-new-team/dialog-new-team.component.html [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    <input matInput placeholder="Nazwisko" formControlName="lastName">
+  </mat-form-field>
+  <mat-form-field>
+    <input matInput placeholder="email" formControlName="email">
+  </mat-form-field>
+ </form>
+</div>
+<div mat-dialog-actions>
+  <button mat-button (click)="onSubmit()" [disabled]="!teamFormGroup.valid" tabindex="2">Zapisz</button>
+  <button mat-button (click)="onNoClick()" tabindex="-1">Anuluj</button>
```
Comment (treformat) [inline src/Client/src/app/settings/pages/teams/dialog-new-team/dialog-new-team.component.html]: Same, cancel

File: src/Client/src/app/settings/pages/teams/dialog-new-team/dialog-new-team.component.ts [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+import { TeamService } from '../../../../shared/services/team.service';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'app-dialog-new-team',
+  templateUrl: './dialog-new-team.component.html',
+  styleUrls: ['./dialog-new-team.component.css']
+})
+export class DialogNewTeamComponent implements OnInit {
+  ngOnInit(): void {
```
Comment (treformat) [inline src/Client/src/app/settings/pages/teams/dialog-new-team/dialog-new-team.component.ts]: Reorder, move below constructor

File: src/Client/src/app/settings/pages/teams/teams.component.css [hunk 15]
```diff
@@ -0,0 +1,7 @@
+table {
```
Comment (treformat) [inline src/Client/src/app/settings/pages/teams/teams.component.css]: use previously added "full-width" class

File: src/Client/src/app/shared/models/divisionTeam.viewmodel.ts [hunk 16]
```diff
@@ -0,0 +1,6 @@
+export interface IDivisionTeam {
+    id: string;
+    email: string;
+    FfllTeamName: string;
```
Comment (treformat) [inline src/Client/src/app/shared/models/divisionTeam.viewmodel.ts]: WTF :)
FullTeamName

File: src/Client/src/app/shared/services/team.service.ts [hunk 17]
```diff
@@ -18,13 +20,31 @@ export class TeamService {
...
+    return this.httpClient.get<ITeam[]>(getTeamUrl);
+  }
+
+  filterByName(name: any): Observable<ITeam[]> {
+    const getTeamsUrl = this.baseUrl + 'team/'+ name;
+    
+    return this.httpClient.get<ITeam[]>(getTeamsUrl);
+  }
+
+  getByDivision(divisionId: string): Observable<IDivisionTeam[]> {
```
Comment (treformat) [inline src/Client/src/app/shared/services/team.service.ts]: I would move it to DivisionTeamService

File: src/Client/src/app/settings/pages/teams/teams.component.ts [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+export interface PeriodicElement {
+  name: string;
+  position: number;
+  weight: number;
+  symbol: string;
+}
+
+const ELEMENT_DATA: PeriodicElement[] = [
```
Comment (treformat) [inline src/Client/src/app/settings/pages/teams/teams.component.ts]: I would return this from API

**Comment by pmozola** [comment]
needs work :)

**Pull request closed without merging**